### PR TITLE
BlitBuffer: add :hatchRect()

### PR DESCRIPTION
**Pull request by poire-z**
Also fix BB_GET_PIXEL with proper parens around parameters that could be complex.

Will allow for fancy stuff in Page Browse. See https://github.com/koreader/koreader/issues/10193#issuecomment-1712537745 and follow ups.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1661)

### CODE REVIEW

File: blitbuffer.c [hunk 1]
```diff
@@ -145,13 +145,13 @@ static const char*
...
+        *pptr = (COLOR*)(bb->data + (y) * bb->stride) + (x); \
     } else if (rotation == 1) { \
-        *pptr = (COLOR*)(bb->data + x * bb->stride) + bb->w - y - 1; \
+        *pptr = (COLOR*)(bb->data + (x) * bb->stride) + bb->w - (y) - 1; \
     } else if (rotation == 2) { \
-        *pptr = (COLOR*)(bb->data + (bb->h - y - 1) * bb->stride) + bb->w - x - 1; \
+        *pptr = (COLOR*)(bb->data + (bb->h - (y) - 1) * bb->stride) + bb->w - (x) - 1; \
     } else if (rotation == 3) { \
-        *pptr = (COLOR*)(bb->data + (bb->h - x - 1) * bb->stride) + y; \
+        *pptr = (COLOR*)(bb->data + (bb->h - (x) - 1) * bb->stride) + (y); \
```
Comment (benoit-pierre) [inline blitbuffer.c:154]: Why not fix the other arguments too?

Comment (poire-z) [reply blitbuffer.c:154]: Not touching other people stuff other than what they allowed me to :) https://github.com/koreader/koreader/issues/10193#issuecomment-1741879785

**Review by NiLuJe — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1661)* status: all files reviewed, 3 unresolved discussions (waiting on @benoit-pierre and @poire-z)

___
*[`blitbuffer.c` line 647 at r1](https://reviewable.io/reviews/koreader/koreader-base/1661#-NfrGf-r49S0VkOlt1QB:-NfrGf-r49S0VkOlt1QC:b4sx6b9) ([raw file](https://github.com/koreader/koreader-base/blob/943f5447ca2384bdeb079aa160786e9c8ad4439f/blitbuffer.c#L647)):*
> ```C
>                 }
>             }
>             else {
> ```

Style nit: `} else {` ;).

(For all of 'em).
___
*[`blitbuffer.c` line 748 at r1](https://reviewable.io/reviews/koreader/koreader-base/1661#-NfrH97o22yC0YCwuzYl:-NfrH97o22yC0YCwuzYm:bq14b7t) ([raw file](https://github.com/koreader/koreader-base/blob/943f5447ca2384bdeb079aa160786e9c8ad4439f/blitbuffer.c#L748)):*
> ```C
>                             dstptr->g = (uint8_t) color->a;
>                             dstptr->b = (uint8_t) color->a;
>                             // dstptr->alpha = 0xFF;
> ```

Don't ask me why the `rect` painters leave the alpha untouched ;D.

**Review by NiLuJe — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1661)* status: all files reviewed, 3 unresolved discussions (waiting on @benoit-pierre and @poire-z)

**Pull request merged**